gql/transport/phoenix_channel_websockets.py

```diff
@@ -0,0 +1,244 @@
+import asyncio
+import json
+from typing import Dict, Optional, Tuple
+
+from graphql import DocumentNode, ExecutionResult, print_ast
+from websockets.exceptions import ConnectionClosed
+
+from .exceptions import (
+    TransportProtocolError,
+    TransportQueryError,
+    TransportServerError,
+)
+from .websockets import WebsocketsTransport
+
+
+class PhoenixChannelWebsocketsTransport(WebsocketsTransport):
+    def __init__(
+        self, channel_name: str, heartbeat_interval: int = 30, *args, **kwargs
+    ) -> None:
+        self.channel_name = channel_name
+        self.heartbeat_interval = heartbeat_interval
+        self.subscription_ids_to_query_ids: Dict[str, int] = {}
+        super(PhoenixChannelWebsocketsTransport, self).__init__(*args, **kwargs)
```

Comment:
Would be great to document where these comes from. It seems like [Phoenix.Socket](https://hexdocs.pm/phoenix/Phoenix.Socket.html) and [Phoenix.Socket.Transport](https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html) are the keys for this task.

Reply:
Done.